lib/src/rocket.rs

```diff
@@ -214,20 +240,15 @@ impl Rocket {
                 response
             }
             Outcome::Forward(data) => {
-                // Rust thinks `request` is still borrowed here, but it's
-                // obviously not (data has nothing to do with it), so we
-                // convince it to give us another mutable reference.
-                // TODO: Use something that is well defined, like UnsafeCell.
-                // But that causes variance issues...so wait for NLL.
-                let request: &'r mut Request<'s> =
-                    unsafe { (&mut *(request as *const _ as *mut _)) };
 
                 // There was no matching route. Autohandle `HEAD` requests.
                 if request.method() == Method::Head {
                     info_!("Autohandling {} request.", Paint::white("HEAD"));
-                    request.set_method(Method::Get);
-                    let mut response = self.dispatch(request, data);
-                    response.strip_body();
+
+                    // Dispatch the request again with Method `GET`.
+                    request._set_method(Method::Get);
+                    let response = self.route_and_process(request, data);
```

Comment:
Remove the `let` and just return `self.route_and_process(request, data)` directly from this branch.